Voter sees percentage #17

rickbacci opened the pull request:
@joshcheek @selfup @jbrr @rrgayhart 

My goal here is to iterate over the polls keys (split them into the admin/voter strings) and find the one that matches either an admin key, or a voter key.
If i find a match I will build the id (adminString + voterString) and use this to get the poll. I will also set an attribute to know which type of link is being looked at. Then I can show/hide things in the view based on
who it is.

After I have the poll I can pass it to the template.  Thoughts?

rickbacci committed:
add visitor can see percentages test

- add id to h1

rickbacci committed:
move Redis configuration to configureRedis w

rickbacci committed:
rearrange code

rickbacci committed:
create new Poll setter function

rickbacci committed:
create and extract storePoll

rickbacci committed:
extract logging Redis to console

rickbacci committed:
extract host

rickbacci committed:
fix route to polls#show

rickbacci committed:
extract adminString and voterString from id

rickbacci committed:
sudo code for finding the correct poll

rickbacci committed:
update polls#show

selfup commented on server.js:
```diff
@@ -86,4 +80,53 @@ var server = http.createServer(app).listen(port, function () {
+
+function logRedisKeys(redis) {
+  redis.hkeys('polls', function (err, keys) {
+    keys.forEach(function (key) {
```
Looks like you got the iteration down

rickbacci replied:
Holy crap, I forgot about that...lol. Thanks. @selfup

selfup commented on server.js:
```diff
@@ -86,4 +80,53 @@ var server = http.createServer(app).listen(port, function () {
+
+// TODO move Poll to it's own file
+
+function Poll(poll_id, adminString, voterString, request) {
```
Making a class? Interesting!

You going to make methods soon?

rickbacci replied:
I may break it out into prototype methods, i'm thinking the urls should be extracted for sure
@selfup Do i have to put your name in every post to notify you?

selfup replied:
No you are good to go!

I am quite busy right now lol

rickbacci committed:
changed response from redirect to render

rickbacci committed:
rearrange Poll arguments

JoshCheek commented:
Nice job, Ricky! ^_^ Keep kicking ass!

rickbacci committed:
double random string size

rickbacci committed:
extract Poll to it's own file

rickbacci committed:
cleanup templates

rickbacci committed:
extract Poll to it's own file

rickbacci committed:
remove Redis

rickbacci committed:
add iteration over polls

- require loadash
- push poll onto polls
- cleanup polls show
- change to redirect in poll

RickBacci commented:
@joshcheek @selfup @jbrr @rrgayhart

Not having luck with the way i envisioned this. Had a ton of problems with getting the data out of redis, so i scraped it for now. I'm also having problems with the render in show. It works fine on the redirect, but when you click a url the data disappears.

rickbacci committed:
finally can send the correct data

rickbacci committed:
add logic to filter admin/voter links

- set urlType based on what type of link the id
  returned

selfup commented:
Do you know why the data disappears?

Have you messed around with updating and seeing if the data comes through?

rickbacci committed:
fix variable name in poll

rickbacci committed:
remove todo from index

rickbacci committed:
move jQuery to bottom of body

rickbacci committed:
add table for choices 

- add span with percentages

RickBacci commented:
I need to continue on this track now and implement the socket connection for voting

rickbacci committed:
rename button class

rickbacci committed:
add expectations for feature

- percentage changes if you change answers
- rename button class

rickbacci committed:
update Poll with array for votes and defaults

rickbacci committed:
add socket-io 

- connection-count
- status-message
- disconnect
- clean up variables in client

rickbacci committed:
add event listeners to buttons

rickbacci committed:
refactored for loop

rickbacci committed:
fix syntax for poll data-id

rickbacci committed:
build this.votes hash

rickbacci committed:
add data attributes to poll

rickbacci committed:
change class to id in span

rickbacci committed:
clean up whitespace

rickbacci committed:
add vote selection on client

rickbacci committed:
remove socketId from buttons

rickbacci committed:
add voteCount to poll

rickbacci committed:
votes tally in server

rickbacci committed:
add division for percentage

RickBacci commented:
@joshcheek @selfup @jbrr @rrgayhart

jbrr commented:
This is a pretty huge PR, I think it would take me a while to go through all of it. The poll class looks good though, I like the OO approach! Tests are passing? Things are working as you'd expect? What's your next step?

selfup commented:
Agreed with Jeff. But it is merged. Guess the tests passed! What is the next step?

RickBacci commented:
yeah but there is something wierd going on with the data. I need to figure
out a way to clear it after the tests.

On Tue, Jan 19, 2016 at 9:58 AM, Jeff Ruane <notifications@github.com>
wrote:

> This is a pretty huge PR, I think it would take me a while to go through
> all of it. The poll class looks good though, I like the OO approach! Tests
> are passing? Things are working as you'd expect? What's your next step?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/RickBacci/TwoCentsWorth/pull/20#issuecomment-172917072>
> .
>

jbrr commented:
I'm not sure how Nightwatch works, but I have a beforeEach function in my tests that reset the poll object to either a blank object or a fixture, depending on the case. Would something like that be possible?

RickBacci commented:
@jbrr yeah i might be able to do that but these are integration tests so i
was clicking thru


On Tue, Jan 19, 2016 at 1:27 PM, Jeff Ruane <notifications@github.com>
wrote:

> I'm not sure how Nightwatch works, but I have a beforeEach function in my
> tests that reset the poll object to either a blank object or a fixture,
> depending on the case. Would something like that be possible?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/RickBacci/TwoCentsWorth/pull/20#issuecomment-172976016>
> .
>

jbrr commented:
Yeah, true. I know in Capybara there's a way to reset the session between tests, maybe there's something like that?
